### PR TITLE
Feat/hdlc/v5

### PR DESCRIPTION
**Pull request by victorjulien**
#4586, rebased and comments addressed.

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):
Passed.

### CODE REVIEW

File: src/decode-hdlc.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                   const uint8_t *pkt, uint32_t len)
+{
+    StatsIncr(tv, dtv->counter_hdlc);
+
+    if (unlikely(len < HDLC_HEADER_LEN)) {
+        ENGINE_SET_INVALID_EVENT(p, HDLC_PKT_TOO_SMALL);
+        return TM_ECODE_FAILED;
+    }
+
+    if (unlikely(len > HDLC_HEADER_LEN + USHRT_MAX)) {
```
Comment (catenacyber) [inline src/decode-hdlc.c]: Why do we need this check ?

File: src/decode-hdlc.c [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        case ETHERNET_TYPE_IP:
+            DecodeIPV4(tv, dtv, p, pkt + HDLC_HEADER_LEN,
+                       len - HDLC_HEADER_LEN);
+            break;
+        case ETHERNET_TYPE_IPV6:
+            DecodeIPV6(tv, dtv, p, pkt + HDLC_HEADER_LEN,
+                       len - HDLC_HEADER_LEN);
+            break;
+        default:
+            SCLogDebug("p %p pkt %p ether type %04x not supported", p,
```
Comment (catenacyber) [inline src/decode-hdlc.c]: Should we set an event ? Are there no other ether types such as VLAN here ?

**Comment by victorjulien** [comment]
SV OISF/suricata-verify#181

File: src/decode-hdlc.c (hunk 1 above)
Comment (victorjulien) [reply src/decode-hdlc.c]: See https://github.com/OISF/suricata/pull/4586#discussion_r383956184 from the last PR.

**Comment by jasonish** [comment]
Do we have to worry about the other non-Cisco HDLC? A peak at my 2016 PR used CHDLC just in case, but I'm not sure if its ever used in the wild.

File: src/decode-hdlc.c (hunk 2 above)
Comment (victorjulien) [reply src/decode-hdlc.c]: I don't know, I assume so. I've done a different approach in #4613

Comment (victorjulien) [reply src/decode-hdlc.c]: I don't know, I assume so. I've done a different approach in #4613

**Comment by victorjulien** [comment]
> Do we have to worry about the other non-Cisco HDLC? A peak at my 2016 PR used CHDLC just in case, but I'm not sure if its ever used in the wild.

In #4613 I've added a 'C' prefix to address this.

**Comment by victorjulien** [comment]
Replaced by #4613

**Pull request closed without merging**